opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -77,3 +90,116 @@ def export(self, metrics: Sequence[Metric]) -> MetricExportResult:
 
     def shutdown(self) -> None:
         pass
+
+
+class PeriodicExportingMetricReader(MetricReader):
+    """`PeriodicExportingMetricReader` is an implementation of `MetricReader`
+    that collects metrics based on a user-configurable time interval, and passes the
+    metrics to the configured exporter.
+    """
+
+    def __init__(
+        self,
+        exporter: MetricExporter,
+        export_interval_millis: Optional[float] = None,
+        export_timeout_millis: Optional[float] = None,
+    ) -> None:
+        self._exporter = exporter
+        if export_interval_millis is None:
+            try:
+                export_interval_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_INTERVAL", 60000)
+                )
+            except ValueError:
+                export_interval_millis = 60000
+        if export_timeout_millis is None:
+            try:
+                export_timeout_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_TIMEOUT", 30000)
+                )
+            except ValueError:
+                export_timeout_millis = 30000
+        self._export_interval_millis = export_interval_millis
+        self._export_timeout_millis = export_timeout_millis
+        self._meter_provider = None  # type: MeterProvider
+        self._shutdown = False
+        self._daemon_thread = threading.Thread(
+            target=self.collect, daemon=True
+        )
+        self._condition = threading.Condition()
+        self._flush_event = None  # type: Optional[threading.Event]
+        self._lock = threading.Lock()
+        self._daemon_thread.start()
+        if hasattr(os, "register_at_fork"):
+            os.register_at_fork(
+                after_in_child=self._at_fork_reinit
+            )  # pylint: disable=protected-access
+
+    def _at_fork_reinit(self):
+        self._condition = threading.Condition()
+        self._lock = threading.Lock()
+        self._daemon_thread = threading.Thread(
+            target=self.collect, daemon=True
+        )
+        self._daemon_thread.start()
+
+    def _register_meter_provider(self, provider: MeterProvider) -> None:
+        self._meter_provider = provider
+
+    def collect(self) -> None:
+        while not self._shutdown:
+            with self._condition:
+                # might have received flush/shutdown request
+                if not self._flush_pending and not self._shutdown:
+                    self._condition.wait(self._export_interval_millis / 1e3)
```

Comment:
Can this division be moved to the moment when the value of `self._export_interval_millis` (would need this attribute to be renamed to `self._export_interval_seconds`) is set to avoid doing it over and over again?

Reply:
Moved outside the loop.